js_methods.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/StringLiterals, Style/AndOr
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield to_a[i] if block_given?
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield to_a[i], i if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    temp_arr = []
+    my_each do |e|
+      temp = yield e
+      temp_arr << e if temp
+    end
+    temp_arr
+  end
+
+  def my_all?
+    ans = true
+    my_each do |e|
+      ans = false unless yield e
+    end
+    ans
+  end
+
+  def my_any?
+    ans = false
+    my_each do |e|
+      ans = true if yield e
+    end
+    ans
+  end
+
+  def my_none?
+    ans = true
+    my_each do |e|
+      ans = false if yield e
+    end
+    ans
+  end
+
+  def my_count(*chr)
+    # why can't use *char directly???
+    cnt = 0
+    char = *chr
+    if !char.empty?
+      my_each { |e| cnt += 1 if e == char[0] }
+      cnt
+    else
+      length
+    end
+  end
+
+  def my_map
+    arr = []
+    if block_given?
+      my_each do |e|
+        elm = yield e
+        arr << elm
+      end
+      arr
+    else
+      to_enum
+    end
+  end
+
+  def my_inject(*init)
+    init, sym = *init
+    init.is_a?(Integer) ? acc = init : sym = init
+    my_each do |e|
+      acc = e and next if acc.nil? # to not sum 2times 1st value
+      acc = if block_given? # because if and else asign to the same variable
+              yield acc, e
+            else
+              acc.send(sym, e)
+            end
+    end
+    acc
+  end
```

Comment:
**[REQUIRED]**
* Combines all elements of enum by applying a binary operation, specified by a block.
* when a symbol is specified combines each element of the collection by applying the symbol as a named method